File: Framework/backend-axum/src/utils/result.rs
use serde::Serialize;

// JSON返回内容的统一封装
#[derive(Debug, Serialize)]
pub struct Result<T> {
    code: i32,
    message: String,
    data: T,
}

impl<T> Result<T> {
    pub fn success(_data: T) -> Result<T> {
        let result = Result {
            code: 200,
            message: String::from("success"),
            data: _data,
        };
        return result;
    }
    pub fn fail(_code: i32, _message: String, _data: T) -> Result<T> {
        let result = Result {
            code: _code,
            message: _message,
            data: _data,
        };
        return result;
    }
}

File: Framework/fullstacknext/src/lib/result.ts
// 返回类型的接口

export interface Result {
	code: number,
	message?: string,
	data?: any;
};

export function succes(data: any): Result {
	const result = setResult({
		data: data
	});
	return result;
}

export function fail(code: number, message: string): Result {
	const result = setResult({
		code: code,
		message: message
	});
	return result;
}

interface SetResultParams {
	code?: number;
	message?: string;
	data?: any;
}

const setResult = (params: SetResultParams = {}): Result => {
	return {
		code: params.code || 200,
		message: params.message || "success",
		data: params.data || null,
	};
};
